diff --git a/pyanaconda/payload/dnf/payload.py b/pyanaconda/payload/dnf/payload.py
index 5312825..db48437 100644
--- a/pyanaconda/payload/dnf/payload.py
+++ b/pyanaconda/payload/dnf/payload.py
@@ -149,6 +149,16 @@ class DNFPayload(Payload):
 
     def _set_additional_repos_from_opts(self, opts):
         """Set additional repositories based on the Anaconda options."""
+           
+        repo = RepoConfigurationDataConf()
+        repo.name = "installer.repo"
+        repo.enabled = True 
+        repo.type = URL_TYPE_BASEURL
+        repo.url = "file:///tmp/installer.repo"
+        repo.installation_enabled = True
+        ks_repo = convert_repo_data_to_ks_repo(repo)
+        self.data.repo.dataList().append(ks_repo)
+        
         for repo_name, repo_url in opts.addRepo:
             try:
                 source = SourceFactory.parse_repo_cmdline_string(repo_url)
